site-scrapers/CVS.js

```diff
@@ -0,0 +1,47 @@
+const sites = require("../data/sites.json");
+const https = require("https");
+const crypto = require("crypto");
+const { map } = require(".");
+
+const siteName = "CVS";
+const site = sites[siteName];
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log(`${siteName} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${siteName} done.`);
+    return webData.map((responseLocation) => {
+        let hasAvailability = parseInt(responseLocation.totalAvailable)
+            ? true
+            : false;
+        let availability = responseLocation.totalAvailable;
+        return {
+            name: `${siteName} (${responseLocation.city})`,
+            hasAvailability,
+            availability,
+            signUpLink: site.website,
+            ...responseLocation,
+        };
+    });
+};
+
+async function ScrapeWebsiteData(browser) {
+    let rawData = {};
+    const page = await browser.newPage();
+    await page.goto(site.website, { waitUntil: "domcontentloaded" });
+    await page.waitForSelector("a[data-modal='vaccineinfo-MA']");
+
+    const [searchResponse, ...rest] = await Promise.all([
+        Promise.race([
+            page.waitForResponse(
+                "https://www.cvs.com/immunizations/covid-19-vaccine.vaccine-status.ma.json?vaccineinfo"
+            ),
+            page.waitForNavigation(),
```

Comment:
Err…I am confused why we need to `Promise.race()` and then `waitForNavigation` if we are retrieving 2 pages in sequence.

Reply:
Good call. Simplified that logic.